gammapy/maps/wcsnd.py

```diff
@@ -231,21 +234,52 @@ def reduce_over_axes(self, func, keepdims=False):
         keepdims : bool, optional
             If this is set to true, the axes which are summed over are left in
             the map with a single bin
+        axes: list
+            Names of MapAxis to reduce over
+            If None, all will reduced
 
         Returns
         -------
-        map_out : WcsNDMap
+        map_out : ~WcsNDMap
             Map with non-spatial axes reduced
         """
-        axis = tuple(range(self.data.ndim - 2))
-        geom = self.geom.to_image()
+        if axes is None:
+            axes = [ax.name for ax in self.geom.axes]
+
+        map_out = self.copy()
+        for ax in axes:
+            map_out = map_out.reduce(ax, keepdims=keepdims)
```

Comment:
It seem like `func` is not passed here...this might be related to the Travis fails we see.